pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -795,4 +796,40 @@ protected void validatePersistencePolicies(PersistencePolicies persistence) {
                         persistence.getBookkeeperAckQuorum()));
 
     }
+
+    protected CompletableFuture<SchemaCompatibilityStrategy> getSchemaCompatibilityStrategyAsync() {
+        validateTopicOperation(topicName, TopicOperation.GET_SCHEMA_COMPATIBILITY_STRATEGY);
+
+        CompletableFuture<SchemaCompatibilityStrategy> future =
+                CompletableFuture.completedFuture(SchemaCompatibilityStrategy.UNDEFINED);
+
+        if (config().isTopicLevelPoliciesEnabled()) {
+            future = getTopicPoliciesAsyncWithRetry(topicName).thenApply(
+                    op -> op.map(TopicPolicies::getSchemaCompatibilityStrategy)
+                            .orElse(SchemaCompatibilityStrategy.UNDEFINED));
+        }
+
+        return future.thenApply(schemaCompatibilityStrategy -> {
+                    if (schemaCompatibilityStrategy == SchemaCompatibilityStrategy.UNDEFINED) {
+                        Policies policies = getNamespacePolicies(namespaceName);
+                        if (policies.schema_compatibility_strategy == SchemaCompatibilityStrategy.UNDEFINED) {
+                            schemaCompatibilityStrategy = SchemaCompatibilityStrategy.fromAutoUpdatePolicy(
+                                    policies.schema_auto_update_compatibility_strategy);
+                            if (schemaCompatibilityStrategy == SchemaCompatibilityStrategy.UNDEFINED) {
+                                return config().getSchemaCompatibilityStrategy();
```

Comment:
The `schema_auto_update_compatibility_strategy` is never equal to `SchemaCompatibilityStrategy.UNDEFINED`, so broker config is always ignored.

Reply:
Please check the old logic carefully.
`schema_auto_update_compatibility_strategy` has lower priority than broker config.
See org.apache.pulsar.broker.admin.impl.NamespacesBase#internalGetSchemaCompatibilityStrategy.

Reply:
If the broker config is `SchemaCompatibilityStrategy .UNDEFINED`, should I set the `schema_auto_update_compatibility_strategy` to broker level?